src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -92,21 +156,21 @@ static int execute_host_command(
     int32_t* required_buffer_size)
 {
     pal::dll_t corehost;
-    hostpolicy_contract host_contract{};
+    hostpolicy_contract_t hostpolicy_contract{};
     corehost_main_with_output_buffer_fn host_main = nullptr;
 
-    int code = load_hostpolicy(impl_dll_dir, &corehost, host_contract, "corehost_main_with_output_buffer", &host_main);
+    int code = load_hostpolicy(impl_dll_dir, &corehost, hostpolicy_contract, "corehost_main_with_output_buffer", &host_main);
```

Comment:
Just a worry:
I guess this already exists as a problem today... but still.
If somebody calls this - it will load hostpolicy.
After that it's possible to call some other entry point which will also load hostpolicy, potentially a different one.
If it's the same one, we're probably fine since this calls unload.
If it's a different one, not sure how OS handles it... maybe it just works (two different dlls with the same name)

In any case, since this doesn't perform any synchronization we rely on the caller to not mix/match.

Probably a non-issue... just wanted to raise this for awareness.

Reply:
I added a log line for if hostpolicy was already loaded, the directory it was loaded from, and the requested directory (but did not change the existing behaviour).